### PR TITLE
Use Mongo 3.4 for 64-bit platforms.

### PR DESCRIPTION
**Pull request by abernix**
Most of the work to prepare for this change was done through the
excellent work of @hwillson in meteor/meteor#9173 which, after some
re-working to support the 64-bit architecture on Windows platforms,
landed in meteor/meteor#9218, making this change as simple as bumping
the minor version number (and rebuilding the dev bundle).

From this point forward, and due to Mongo's discontinuation of 32-bit
support in newer versions of MongoDB, 64-bit platforms will, in
development, use newer versions of Mongo while 32-bit architectures
will remain at 3.2.x versions.

Of course, in production, apps are free to use whichever version of
Mongo they would like, provided that version is supported by the
Node Mongo driver and Meteor's Mongo data packages.  At this time
there is no target for when Meteor will stop supporting Mongo 3.2,
but developers are encouraged to take steps to upgrade their Mongo
deployments (through their database providers) to newer versions
since Mongo has set September 2018 as the "End-of-Life" for Mongo
3.2.x.  For more information on Mongo support cycles, see their
support documents at https://www.mongodb.com/support-policy.

Refs: https://github.com/meteor/meteor/pull/9173
Refs: https://github.com/meteor/meteor/pull/9218

### CODE REVIEW

**Comment by abernix** [comment]
Test/CI failures are due to https://github.com/babel/babel/issues/6824.

tl;dr: Babel published a `babel@7.0.0-bridge.0` package, which takes npm version precedence over the `babel@7.0.0-beta.0` we're relying on (yes, as in, `"beta" < "bridge"`, alphabetically).

```
Error: Cannot find module '@babel/core'
```

**Comment by abernix** [comment]
@benjamn I'm likely jumping the gun by assuming `meteor-babel@7.0.0-beta.3-2` was ready, but if it's any proof that the change you made in that version via https://github.com/meteor/babel/commit/ac6f3bef07416f23f189c7bce8f37eb26c8b5875 worked, I rebuilt the dev bundle in this PR with that version bump (https://github.com/meteor/meteor/pull/9396/commits/109b1116ac581921b322ed9eb4913db7307ea707) and it is now passing.

**Comment by benjamn** [comment]
@abernix My only hesitation about that version of `meteor-babel` is that [tests aren't really passing in Node 4](https://travis-ci.org/meteor/babel/jobs/304843594), but that doesn't really matter I suppose.

**Comment by abernix** [comment]
Since I've intentionally force pushed this PR to exclude the (test) dev bundle bump version, this will now fail again, but the previous result was a success so I'll merge this to `devel`, and rebuild the dev bundle (with a non-test version) on `devel`.
